### ISSUE TITLE
Delete removed commit history

### ISSUE THREAD

**Issue by JokerQyou** [gogits/gogs#3504]
- Gogs version (or commit ref): 0.9.60.0803
- Git version: 1.9.1
- Operating system: Ubuntu 12.04
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

I accidentally added some sensitive data into the source files of one of my repositories ( you can think that I hard-coded my password in the source file). And I removed them from the history by using [bfg](https://rtyley.github.io/bfg-repo-cleaner/) (it's the same as `git-filter-branch` as far as I can tell), and then I did a `git push -f` to my gogs site. After that I cannot find any trace of the removed data from my repository, so far so good.

But then I noticed the old commits are still available on the home page, and by click on the link of these commits, I can still browse the full file tree at that time. By doing so I can still see the removed data. I suppose this is because these data was updated to gogs' database upon each `git push`, and would not be altered afterwards.

Now how can I totally remove these links so that no one can see the removed data from the web interface?

Additional information: when I remove the sensitive data from git commit history, I followed the guide on Github help documentation [here](https://help.github.com/articles/remove-sensitive-data/) and bfg guide [here](https://rtyley.github.io/bfg-repo-cleaner/).

**Comment by bkcsoft** [comment]
Run `git gc` in the repos settings and it should go byebye :)

**Comment by Unknwon** [comment]
Yes, we have `gc` option in admin panel.

**Comment by JokerQyou** [comment]
No. I did run that `git gc` task, actually twice. Maybe I did not make myself clear. I'm not saying that the sensitive data is still in my repo, I'm saying it's still on the web interface.

I can see all the old commits from this page:
![image](https://cloud.githubusercontent.com/assets/1465267/17915517/d5f224e8-69dd-11e6-84ee-ed73260c488f.png)

The commits in the green outline are these I pushed with `git push -f`, which are new ones. These in the red outline are the old commits. (Yes I did it twice to use `bfg` tool right)
![image](https://cloud.githubusercontent.com/assets/1465267/17915559/138c342e-69de-11e6-9a3e-1eb44f49049a.png)

Although I can see these old commits I don't think it's a problem, I thought they are no longer valid links. But when I click on these links, the page is actually accessible, and I can just browse all the source code at that time point. The sensitive data I removed is still there, not in the repo, bu on the web interface of gogs.

Now this is what I was asking, I want to remove these from the web interface so that no one could ever see them.
